### ISSUE TITLE
vmstroage restart

### ISSUE THREAD

**Issue by dw-3** [VictoriaMetrics/VictoriaMetrics#3833]
### Describe the bug

vmstorage restart,no oom

### To Reproduce

![image](https://user-images.githubusercontent.com/20983001/219536032-da98589f-9dd2-47ea-b14b-4e8106211cb3.png)

### Version

vminsert-20221110-224751-tags-v1.83.1-cluster-0-g6d984cdef

### Logs

![image](https://user-images.githubusercontent.com/20983001/219535934-a8a2687d-9f83-4d46-a82b-a429cb905fb0.png)

**Comment by dw-3** [comment]
<img width="1391" alt="image" src="https://user-images.githubusercontent.com/20983001/219848486-38711c8b-2db7-4da9-9257-2844035a2e49.png">

**Comment by dw-3** [comment]
2023-02-17T07:53:19.530+0800	info	VictoriaMetrics/lib/memory/memory.go:42	limiting caches to 1288490188 bytes, leaving 858993460 bytes to the OS according to -memory.allowedPercent=60

**Comment by zekker6** [comment]
@dw-3 According to the log from the first screenshot vmstorage wasn't able to open a file which was supposed to be in place.
This might be caused by some external process clearing "tmp" files, disk corruption, underlying file system issues or other reasons. When vmstorage detects such issue it does exit with an error code since data cannot be safely read/written anymore.
Please, check if file system/hard drives are not affectd and that there are not collisions in used software data storage folders.

**Comment by dw-3** [comment]
There is one thing I need to explain. This vmstorage has been restarted twice. The log in the above screenshot is the log of the latest restart, but the log before the first restart has been overwritten, so I cannot find the reason for the first restart. There is no possibility that the process exited abnormally during the first restart, and then the tmp file was not found after the second restart, and the log of the second restart we saw is just the appearance, the first restart is the root cause, I am now I collected the logs of vmstorage with elk. If there is such a phenomenon again, I can capture the logs. The vmstorage restarted twice this time, resulting in data loss for nearly 2 minutes. In the cluster mode of vm, adjust the replicationFactor to 2, which can Can't achieve high availability of vmstorage to avoid the occurrence of the above situation?

**Comment by zekker6** [comment]
> If the replicationFactor is adjusted to 2 in the cluster mode of vm, it can Can't achieve high availability of vmstorage to avoid the occurrence of the above situation?

Yes. Having replication factor set to 2 will mean that vminsert will put every data point at two different storage nodes. In case any of these nodes will fail the other one will still have replica of same data available.

**Comment by dw-3** [comment]
> > If the replicationFactor is adjusted to 2 in the cluster mode of vm, it can Can't achieve high availability of vmstorage to avoid the occurrence of the above situation?
> 
> Yes. Having replication factor set to 2 will mean that vminsert will put every data point at two different storage nodes. In case any of these nodes will fail the other one will still have replica of same data available.

I understand that writing data to two different vmstorages is to achieve high availability
